packages/asset/test/AssetCreate.test.ts

```diff
@@ -1572,4 +1572,576 @@ describe('AssetCreate (/packages/asset/contracts/AssetCreate.sol)', function ()
+        expect(balanceAfter).to.be.lt(balanceBefore);
+      });
+      it('should increment the creator nonce', async function () {
+        const {
+          mintCatalyst,
+          lazyMintAsset,
+          generateLazyMintSignature,
+          metadataHashes,
+          creator,
+          AssetCreateContract,
+        } = await runCreateTestSetup();
+        await mintCatalyst(4, 5);
+        const signature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const nonceBefore = await AssetCreateContract.creatorNonces(
+          creator.address
+        );
+        await lazyMintAsset(
+          signature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+        const nonceAfter = await AssetCreateContract.creatorNonces(
+          creator.address
+        );
+        expect(nonceAfter).to.be.gt(nonceBefore);
+      });
+      it('should increase the amount for already minted asset', async function () {
+        const {
+          mintCatalyst,
+          lazyMintAsset,
+          generateLazyMintSignature,
+          metadataHashes,
+          creator,
+          AssetContract,
+        } = await runCreateTestSetup();
+        await mintCatalyst(4, 10);
+
+        const firstMintSignature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const firstMintResult = await lazyMintAsset(
+          firstMintSignature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+
+        const firstEventData = firstMintResult.events.filter(
+          (e: Event) => e.event == 'AssetLazyMinted'
+        )[0].args;
+
+        const secondMintSignature = await generateLazyMintSignature(
+          creator.address,
+          4,
+          5,
+          metadataHashes[0],
+          10
+        );
+        const secondMintResult = await lazyMintAsset(
+          secondMintSignature,
+          4,
+          5,
+          metadataHashes[0],
+          10,
+          creator.address
+        );
+        const secondEventData = secondMintResult.events.filter(
+          (e: Event) => e.event == 'AssetLazyMinted'
+        )[0].args;
+
+        // tokenIds should be the same
+        expect(secondEventData.tokenId).to.equal(firstEventData.tokenId);
+
+        // users balance should be 10
+        expect(
+          await AssetContract.balanceOf(creator.address, firstEventData.tokenId)
+        ).to.equal(10);
+      });
+    });
+    describe('Revert', function () {
+      it('should revert if signature is invalid', async function () {
+        const {lazyMintAsset, metadataHashes, creator} =
+          await runCreateTestSetup();
+        const signature =
```

Comment:
Can you add a note on how this sig is invalid? (Ideally we would set up the sig with a bad input for each param and have multiple unit tests)

Reply:
Right, yeah this is a made up signature, but testing for bad input for each param is the way to go. I will add those.